torch/_dynamo/guards.py

```diff
@@ -506,6 +512,27 @@ def compile_check_fn(
             )
             verbose_code_parts.append(f"___check_tensors_verbose({verbose_args})")
 
+        graphargs = self.output_graph.graphargs
+
+        aotautograd_guards: List[
+            GuardEnvExpr
+        ] = self.output_graph.tracing_context.guards_context.aotautograd_guards
+        for guard in aotautograd_guards:
+            if isinstance(guard, DuplicateInputs):
+                pos_a = guard.input_pos_a
+                pos_b = guard.input_pos_b
+                assert pos_b < len(graphargs) and pos_a < len(
+                    graphargs
+                ), "Deduped args out of bounds"
+                assert graphargs[pos_a].is_tensor, "Deduped arg must be a tensor"
+                assert graphargs[pos_b].is_tensor, "Deduped arg must be a tensor"
```

Comment:
I'm not hot on using index positions. The strongest argument against it, I think, is that we need to account for duplicates in parameters too, but this is unrepresentable in the current formulation.

To use not-indexes, we have to propagate source names for the args as well as the parameters/buffers. This is annoying, but doable, I think (maybe we store it sideband in AOTConfig.)

Reply:
Discussed offline. Will do a subsequent PR where we move names to here.